### PR TITLE
SSL certificate is fetched on startup

### LINKED ISSUE #60: SSL certificate is fetched on first request only
**Describe the bug**

The SSL certificate is not fetched right on cli start, but only when the first request is coming. I would like to have a fast response time of the first request already.

**To Reproduce**
1. Host a loophole site
2. Visit the page
4. See the response after a long time only

**Expected behavior**
The site should not be available before the certificate is successfully fetched.

**Screenshots**
N/A

**Environment**
 - OS: macOS
 - Version `loophole version 1.0.0-beta.7 (95033a34197863fe1074bf148758b30d16477fb6)`

**Additional context**
I am also fine with leaving it as is, but would like to have the pros and cons publicly discussed so we can link to this discussion if the question pops up again.

### PR DESCRIPTION
**Pull request by Morishiri**
Fixes #60

### CODE REVIEW

File: internal/app/loophole/loophole.go [hunk 1]
```diff
@@ -378,7 +378,15 @@ func Start(config lm.Config) {
 
 	listenerHTTPSOverSSH, proxiedEndpointHTTPS, siteSpecs := generateListener(config, publicKeyAuthMethod, publicKey, siteSpecs)
 	defer listenerHTTPSOverSSH.Close()
-
+	go func() {
+		log.Debug().Msg("Issuing request to provision certificate")
```
Comment (0x7f) [inline internal/app/loophole/loophole.go:382]: Maybe better to make this `Info` as well so the user knows what is going on right now.

File: internal/app/loophole/loophole.go [hunk 2]
```diff
@@ -378,7 +378,15 @@ func Start(config lm.Config) {
 
 	listenerHTTPSOverSSH, proxiedEndpointHTTPS, siteSpecs := generateListener(config, publicKeyAuthMethod, publicKey, siteSpecs)
 	defer listenerHTTPSOverSSH.Close()
-
+	go func() {
```
Comment (0x7f) [inline internal/app/loophole/loophole.go:381]: Stupid go question. Why put this into an anonymous function that is called in-place? Why not move it to a dedicated function and call it here, or just put the code right in place, as i don't see any conflicting variables or sth like that.

**Review by 0x7f — APPROVED** [review]

File: internal/app/loophole/loophole.go (hunk 2 above)
Comment (Morishiri) [reply internal/app/loophole/loophole.go:381]: It's go-routine which allows you to achieve more-or-less threading. I don't want to block the code here

Comment (Morishiri) [reply internal/app/loophole/loophole.go:381]: About extraction of this, I'm just working on it as a separate restructure PR. This file became too big and ugly. I want to extract certain parts to separate modules to make it nicer

Comment (0x7f) [reply internal/app/loophole/loophole.go:381]: Ah, okay, didn't know that it is async this way. Tbh, I would expect this to be blocking, though. I thought that's the point of this PR. That when the cli prints the domain, it is ready to use and the first commit is already quick, including a valid certificate.

Comment (Morishiri) [reply internal/app/loophole/loophole.go:381]: If I won't make it async it cannot accept the connection which is sent, that's chicken and egg more-or-less

Comment (Morishiri) [reply internal/app/loophole/loophole.go:381]: Unless I make them both go routines and stuck the main process in waiting mode, that may be an option if you want to achieve it :)

**Review by denny241 — APPROVED** [review]
I think it's good the way it is right now. 
Slowing down the CLI output to wait for the certificate to be done, or something like that, would only make the user experience better at the cost of making it worse imo, so there wouldn't really be any net gain.

Right now it might not be done fetching the certificate in 100% of the cases before the user opens or shares the link, but especially for first time users who have to process the information we provide them before taking action it should have the cert ready by the time they open the link.

**Comment by Morishiri** [comment]
Contains changes from #75 so has to eb merged after.

**Pull request merged**